### PR TITLE
users registration changes

### PR DESCRIPTION
**Pull request by ShubhamKhandelwal7**
added sign-in sign-up related functionality.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class ApplicationController < ActionController::Base
+  before_action :authorize
+
+  def authorize
```
Comment (Anuragjain89) [inline app/controllers/application_controller.rb:4]: Please make this protected.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,2 +1,9 @@
...
+    unless User.find_by(password_reset_token: cookies[:password_reset_token])
```
Comment (Anuragjain89) [inline app/controllers/application_controller.rb:5]: Why are we using password_reset_token for authentication / authorisation.

File: app/controllers/password_resets_controller.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+class PasswordResetsController < ApplicationController
+  def new
```
Comment (Anuragjain89) [inline app/controllers/password_resets_controller.rb:2]: You will need to skip authentication / authorisation callbacks for public actions.

File: app/controllers/password_resets_controller.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    end
+  end
+
+  def update
+    @user = User.find_by(password_reset_token: params[:id])
+    if @user.password_reset_sent_at < 2.hours.ago
+      redirect_to :new, notice: "Password token expired, please try again"
+    elsif @user.update(user_params)
+      redirect_to login_url, notice: "Password reset success"
+    else
```
Comment (Anuragjain89) [inline app/controllers/password_resets_controller.rb:25]: Can we extract this logic related to expiration into model ?

File: lib/tasks/create_admin.rake [hunk 5]
```diff
@@ -0,0 +1,15 @@
+namespace :admin do
```
Comment (Anuragjain89) [inline lib/tasks/create_admin.rake:1]: Good Work 👍 

While it's good that you know how to create a rake task and that rake task is nothing but just another process to execute your ruby code, this one is not the exact use case.

We generally seed data via seed file or rails console.So please remove this task.

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  def verify
+    user = User.find(params[:id])
+
+    respond_to do |format|
+      if user
+        format.html { redirect_to login_url, notice: "User successfully verified" }
+      else
+        format.html { redirect_to login_url, notice: "User could not got verified" }
+      end
+    end
```
Comment (Anuragjain89) [inline app/controllers/users_controller.rb:28]: As discussed, please use a token.
Do not use notice for error messages.
Avoid redundant format.html.

File: app/models/user.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    password_reset_sent_at < 2.hours.ago
+  end
+
+  private
+
+  def generate_token(column)
+    self[column] = SecureRandom.urlsafe_base64
+    rescue StandardError 
+      TOKEN_ERROR_MSG
+  end
```
Comment (Anuragjain89) [inline app/models/user.rb:35]: How do we assure uniqueness for this ?

File: app/models/user.rb [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+
+  def send_password_reset
+    generate_token(:password_reset_token)
+    self.password_reset_sent_at = Time.now
+    save!(validate: false)
+    UserMailer.password_reset(self).deliver
+  end
+
+  def password_reset_token_expired?
```
Comment (Anuragjain89) [inline app/models/user.rb:25]: This should not be public

File: app/models/user.rb [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    password_reset_sent_at < 2.hours.ago
```
Comment (Anuragjain89) [inline app/models/user.rb:26]: Please move this to constant.

File: app/controllers/sessions_controller.rb [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+  def create
+    user = User.find_by(email: params[:email])
+    if user && user.authenticate(params[:password])
+      if params[:remember_me]
+        cookies.permanent[:user_id] = user.id
+      else 
+        cookies[:user_id] = user.id
+      end
+      # redirect to homepage
```
Comment (Anuragjain89) [inline app/controllers/sessions_controller.rb:15]: Please replace this comment with a TODO comment.

File: .gitignore [hunk 11]
```diff
@@ -1,4 +1,41 @@
...
+/public/packs-test
+/node_modules
+/yarn-error.log
+yarn-debug.log*
+.yarn-integrity
+
+/test
+
+# Ignore application configuration
+/config/application.yml
```
Comment (bansalakhil) [inline .gitignore:41]: @ShubhamKhandelwal7 database.yml should also e in git ignore

**Pull request closed without merging**